src/client/mod.rs

```diff
@@ -75,6 +77,11 @@ impl<C: NetworkConnector> Client<C> {
         self.redirect_policy = policy;
     }
 
+    /// Set the UserAgent. Pass `None` to disable.
+    pub fn set_user_agent(&mut self, agent: Option<String>) {
```

Comment:
Perhaps. Still, that's a different approach than this PR. I'm going to close this.